src/dynamics/video_player/player/states.js

```diff
@@ -845,16 +845,22 @@ Scoped.define("module:VideoPlayer.Dynamics.PlayerStates.PlayVideo", [
                 } else {
                     if (Math.abs(this.dyn.getCurrentPosition() - this.dyn.get("duration")) < 0.1) {
                         this.next("PostrollAd");
-                    } else this.next("MidrollAd");
+                    } else {
+                        if (this.dyn.get("duration") > this.dyn.get("max_shortform_video_duration")) {
+                            this.next("MidrollAd")
+                        }
+                    };
                 }
             }, this);
             this.listenOn(this.dyn, "playnextmidroll", function() {
-                if (!this.dyn.get("adsplayer_active")) {
-                    this.dyn.set("adsplayer_active", true);
+                if (this.dyn.get("duration") > this.dyn.get("max_shortform_video_duration")) {
+                    if (!this.dyn.get("adsplayer_active")) {
+                        this.dyn.set("adsplayer_active", true);
+                    }
+                    this.listenOnce(this.dyn.channel("ads"), "adsManagerLoaded", function() {
+                        this.dyn.channel("ads").trigger("load");
+                    });
```

Comment:
fix: Both of these changes don't look right to me because the event shouldn't be triggering in the first place. Is there a way to prevent the event?

Reply:
the first on need to be there, because the event its something else, but the second you are right, I just push the preventing the event to happening if video content is less than `max_shortform_video_duration`